samples/ComputeSharp.SwapChain/Shaders/FourColorGradient.cs

```diff
@@ -0,0 +1,96 @@
+﻿namespace ComputeSharp.SwapChain.Shaders
+{
+    /// <summary>
+    /// A shader which interpolates four different colors into a rotating gradient pattern.
+    /// Ported from <see href="https://www.shadertoy.com/view/7lc3R4"/>.
+    /// <para>Created by Inigo Quilez.</para>
+    /// <para>License Creative Commons Attribution-NonCommercial-ShareAlike 3.0 Unported License.</para>
+    /// </summary>
+    [AutoConstructor]
+    internal readonly partial struct FourColorGradient : IPixelShader<float4>
+    {
+        /// <summary>
+        /// The current time since the start of the application.
+        /// </summary>
+        public readonly float time;
+
+        // Colors to blend together
+        private static readonly float3 colorOne = new(0.999f, 0.999f, 0.999f);
+        private static readonly float3 colorTwo = new(0.999f, 0f, 0f);
+        private static readonly float3 colorThree = new(0f, 0.999f, 0f);
+        private static readonly float3 colorFour = new(0f, 0f, 0.999f);
+
+        private static float2x2 Rotate(float a)
```

Comment:
Nit: full XML comments for all these methods. If they're the same as the ones in other sample shaders you can also just copy-paste them from there.

Reply:
Not all the other samples have this. Also the Rotation one seems really common, why isn't it part of the GLSL api? Does it make sense to have some built-in helpers for these commonly used shader patterns as extensions or something to the `Hlsl` class?